sdk/core/azure-core/inc/azure/core/http/winhttp/win_http_client.hpp

```diff
@@ -6,26 +6,166 @@
+       * every time from HTTP RawResponse.
+       *
+       * @remark This value is also used to avoid trying to read more data from network than what we
+       * are expecting to.
+       *
+       * @remark A value of -1 means the transfer encoding was chunked.
+       *
+       */
+      int64_t m_contentLength;
+
+      int64_t m_streamTotalRead;
+
+    public:
+      WinHttpStream(std::unique_ptr<HandleManager>& handleManager, int64_t contentLength)
+          : m_handleManager(std::move(handleManager)), m_contentLength(contentLength)
+      {
+        this->m_isEOF = false;
+        this->m_streamTotalRead = 0;
+      }
+
+      /**
+       * @brief Implement #BodyStream length.
+       *
+       * @return The size of the payload.
+       */
+      int64_t Length() const override { return this->m_contentLength; }
+
+      /**
+       * @brief Implement #BodyStream read. Calling this function pulls data from the wire.
+       *
+       * @param context #Context so that operation can be canceled.
+       * @param buffer Buffer where data from wire is written to.
+       * @param count The number of bytes to read from the network.
+       * @return The actual number of bytes read from the network.
+       */
+      int64_t Read(Azure::Core::Context const& context, uint8_t* buffer, int64_t count) override;
+    };
+  } // namespace Details
+
+  /**
+   * @brief Sets the WinHTTP session and connection options used to customize the behavior of the
+   * transport.
+   *
+   */
+  struct WinHttpTransportOptions
+  {
+    // Empty struct reserved for future options.
+  };
+
   /**
-   * @brief #HttpTransport implementation via WinHttp.
+   * @brief Concrete implementation of an HTTP transport that uses WinHttp when sending and
+   * receiving requests and responses over the wire.
+   *
    */
-  class WinHttpTansport : public HttpTransport {
+  class WinHttpTransport : public HttpTransport {
   private:
+    WinHttpTransportOptions m_options;
+
+    void CreateSessionHandle(std::unique_ptr<Details::HandleManager>& handleManager);
+    void CreateConnectionHandle(std::unique_ptr<Details::HandleManager>& handleManager);
+    void CreateRequestHandle(std::unique_ptr<Details::HandleManager>& handleManager);
+    void Upload(std::unique_ptr<Details::HandleManager>& handleManager);
+    void SendRequest(std::unique_ptr<Details::HandleManager>& handleManager);
+    void WinHttpTransport::ReceiveResponse(std::unique_ptr<Details::HandleManager>& handleManager);
+    int64_t WinHttpTransport::GetContentLength(
+        std::unique_ptr<Details::HandleManager>& handleManager,
+        HttpMethod requestMethod,
+        HttpStatusCode responseStatusCode);
+    std::unique_ptr<Details::WinHttpStream> GetBodyStream(
+        std::unique_ptr<Details::HandleManager>& handleManager,
+        int64_t contentLength);
+    std::unique_ptr<RawResponse> GetRawResponse(
+        std::unique_ptr<Details::HandleManager>& handleManager,
```

Comment:
```suggestion
        std::unique_ptr<Details::HandleManager> handleManager,
```

Since this is the methid that will handle the ownership from handleManager from the TransportAdapter to the BodyStream, it needs to become the owner of the handleManager. Then, when below, when the bodyStream is created, the unique_ptr needs to be moved to the bodyStream to transfer the ownership